File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stk = []
        for c in s:
            if c in "({[":
                stk.append(c)
            elif c in "}])":
                if not stk:
                    return False
                top = stk[-1]
                if (c == "}" and top == "{") \
                 or (c == "]" and top =="[") \
                 or (c== ")" and top=="("):
                 stk.pop()
                else:
                    return False
               
        return len(stk) == 0

            
         
   


    
        

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
    stack<char>stk;
    for(int i= 0 ; i < s.size(); i++){
    if(s[i] == '(' || s[i] == '{'|| s[i] == '['){
        stk.push(s[i]);
    }
    else if(s[i]==')' || s[i] == '}' || s[i] == ']'){
        if(stk.empty()) return false;

        if(stk.top() == '(' && s[i] == ')' || stk.top() == '{' && s[i] == '}' || stk.top() == '[' && s[i] == ']') {
            stk.pop();
        }
       else{
         return false;} 
    }
       
    }
    
           return stk.empty();

}
};